cmd/peer-rest-server.go

```diff
@@ -1384,6 +1385,82 @@ func (s *peerRESTServer) DriveSpeedTestHandler(w http.ResponseWriter, r *http.Re
 	logger.LogIf(r.Context(), gob.NewEncoder(w).Encode(result))
 }
 
+// DevNull - everything goes to io.Discard
+func (s *peerRESTServer) DevNull(w http.ResponseWriter, r *http.Request) {
+	if !s.IsValid(w, r) {
+		s.writeErrorResponse(w, errors.New("invalid request"))
+		return
+	}
+	for {
+		n, err := io.CopyN(io.Discard, r.Body, 128*1024)
+		atomic.AddUint64(&globalNetperfRX, uint64(n))
+		if err != nil {
+			break
+		}
+	}
+}
+
+// Reader to read random data.
+type netperfReader struct {
+	eof bool
+	n   uint64
+	buf []byte
+}
+
+func (m *netperfReader) Read(b []byte) (int, error) {
+	if m.eof {
+		return 0, io.EOF
+	}
+	n := copy(b, m.buf)
+	atomic.AddUint64(&m.n, uint64(n))
+	return n, nil
+}
+
+func netperf(ctx context.Context, duration time.Duration) madmin.NetperfNodeResult {
+	globalNetperfRX = 0
```

Comment:
Yes there is a small ignorable margin of error to avoid extra code of doing "synchronization step". My tests in lab show that this margin of error is ok to ignore. (i.e shows full 10GBps on the 100gig network)

Reply:
Designs shouldn't rely on mostly working. It will always come back to bite you. I will do proposal that should be easy to do.